default/AI/ShipDesignAI.py

```diff
@@ -1280,6 +1293,22 @@ def _adjusted_production_cost(self):
         # TODO: Rethink about math and maybe work out a more accurate formula
         return self.production_cost**(1/(1+foAI.foAIstate.shipCount * AIDependencies.SHIP_UPKEEP))
 
+    def _effective_fuel(self):
+        """Return the number of turns the ship can move without refueling."""
+        return min(self.fuel / max(1-self.fuel_per_turn, 0.001), 10)
+
+    def _effective_mine_damage(self):
+        return self.additional_specifications.enemy_mine_dmg -self.repair_per_turn
```

Comment:
PEP8: and always have the same amount of whitespace on both sides of a binary operator.